File: frontend/src/features/documents/types/documents.ts
export type DocumentStatus = "uploaded" | "processing" | "ready" | "failed";
export type DocumentSort = "newest" | "oldest" | "title_asc" | "title_desc";
export type DocumentViewMode = "grid" | "list";

export type DocumentItem = {
  id: number;
  title: string;
  originalFilename: string;
  mimeType: string;
  fileSize: number;
  pageCount: number;
  status: DocumentStatus;
  errorMessage: string | null;
  createdAt: string;
  updatedAt: string;
};

export type DocumentListParams = {
  page: number;
  pageSize: number;
  search?: string;
  status?: DocumentStatus | "all";
  sort?: DocumentSort;
};

export type DocumentListResponse = {
  items: DocumentItem[];
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
};

export type UploadDocumentInput = {
  accessToken: string;
  file: File;
  title?: string;
  onProgress?: (progress: number) => void;
};

File: backend/app/schemas/documents.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

DocumentStatus = Literal["uploaded", "processing", "ready", "failed"]
DocumentSort = Literal["newest", "oldest", "title_asc", "title_desc"]


class DocumentRead(BaseModel):
    id: int
    title: str
    original_filename: str
    mime_type: str
    file_size: int
    page_count: int
    status: DocumentStatus
    error_message: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    items: list[DocumentRead]
    page: int = Field(ge=1)
    page_size: int = Field(ge=1, le=50)
    total: int = Field(ge=0)
    total_pages: int = Field(ge=0)


class DocumentUploadResponse(DocumentRead):
    pass
